client/rest-high-level/src/test/java/org/elasticsearch/client/CrudIT.java

```diff
@@ -67,8 +66,29 @@ public void testDelete() throws IOException {
             // Testing non existing document
             String docId = "does_not_exist";
             DeleteRequest deleteRequest = new DeleteRequest("index", "type", docId);
+            ElasticsearchException exception = expectThrows(ElasticsearchException.class,
+                    () -> execute(deleteRequest, highLevelClient()::delete, highLevelClient()::deleteAsync));
+            assertEquals(RestStatus.NOT_FOUND, exception.status());
+            assertEquals("Elasticsearch exception [type=index_not_found_exception, reason=no such index]", exception.getMessage());
```

Comment:
The goal of this test is to check how delete behaves where it doesn't find the document in question. I don't think there is a need to test the index creation on the rest client tests (this is done in the core testing). Just create the index and let the test what it does.